test/resources/call_model_fn_once.py

```diff
@@ -21,6 +21,8 @@ def model_fn(model_dir):
         raise RuntimeError('model_fn called more than once (lock: {})'.format(lock_file))
 
     open(lock_file, 'a').close()
+    # delete lock file otherwise following test run with different python version will fail
+    os.remove(lock_file)
```

Comment:
I believe that removing the lock file here makes the test never fail? I would be okay to skip the tests to py2 like we used to do and avoid that behaviour. Please let me know if that makes sense.

Reply:
You are right! It's probably failing only because I ran the tests of different pyversions consecutively on my own machine. I'll undo the remove, test it and see if we need to skip py2.